plugin/trino-pulsar/src/main/java/io/trino/plugin/pulsar/PulsarColumnHandle.java

```diff
@@ -0,0 +1,255 @@
+    @JsonProperty
+    public boolean isInternal()
+    {
+        return internal;
+    }
+
+    @Override
+    @JsonProperty
+    public String getFormatHint()
+    {
+        return formatHint;
+    }
+
+    @JsonProperty
+    public HandleKeyValueType getHandleKeyValueType()
+    {
+        return handleKeyValueType;
+    }
+
+    @JsonIgnore
+    public boolean isKey()
+    {
+        return Objects.equals(handleKeyValueType, HandleKeyValueType.KEY);
+    }
+
+    @JsonIgnore
+    public boolean isValue()
+    {
+        return Objects.equals(handleKeyValueType, HandleKeyValueType.VALUE);
+    }
+
+    ColumnMetadata getColumnMetadata()
+    {
+        return new PulsarColumnMetadata(name, type, null, null, hidden,
+                internal, handleKeyValueType, new PulsarColumnMetadata.DecoderExtraInfo(
+                mapping, dataFormat, formatHint));
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        PulsarColumnHandle that = (PulsarColumnHandle) o;
+
+        if (hidden != that.hidden) {
+            return false;
+        }
+        if (internal != that.internal) {
+            return false;
+        }
+        if (catalogName != null ? !catalogName.equals(that.catalogName) : that.catalogName != null) {
+            return false;
+        }
+        if (name != null ? !name.equals(that.name) : that.name != null) {
+            return false;
+        }
+        if (type != null ? !type.equals(that.type) : that.type != null) {
+            return false;
+        }
+        if (mapping != null ? !mapping.equals(that.mapping) : that.mapping != null) {
+            return false;
+        }
+        if (dataFormat != null ? !dataFormat.equals(that.dataFormat) : that.dataFormat != null) {
+            return false;
+        }
+
+        if (formatHint != null ? !formatHint.equals(that.formatHint) : that.formatHint != null) {
+            return false;
+        }
+
+        return Objects.equals(handleKeyValueType, that.handleKeyValueType);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int result = catalogName != null ? catalogName.hashCode() : 0;
+        result = 31 * result + (name != null ? name.hashCode() : 0);
+        result = 31 * result + (type != null ? type.hashCode() : 0);
+        result = 31 * result + (hidden ? 1 : 0);
+        result = 31 * result + (internal ? 1 : 0);
+        result = 31 * result + (mapping != null ? mapping.hashCode() : 0);
+        result = 31 * result + (dataFormat != null ? dataFormat.hashCode() : 0);
+        result = 31 * result + (formatHint != null ? formatHint.hashCode() : 0);
+        result = 31 * result + (handleKeyValueType != null ? handleKeyValueType.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public String toString()
+    {
+        return "PulsarColumnHandle{"
```

Comment:
Use `ToStringHelper`, here and everywhere else.

Reply:
sure, updated.